packages/react-ui/src/components/molecules/UploadField.js

```diff
@@ -111,14 +114,26 @@ function UploadField(props) {
     setFilesText('');
   };
 
-  const dragPlaceholderText = dragOver ? props.dragPlaceholder : props.placeholder;
+  const setPlaceholder = () => {
```

Comment:
I would change the naming here, by convention `setFoo` evokes something that will change some `Foo` state somewhere. And this looks more like a `getPlaceHolder` to me. Maybe a better way still is a `useMemo` depending on `multiple` and `dragOver`.